convert2rhel/subscription.py

```diff
@@ -139,34 +142,78 @@ def hide_password(cmd):
+        return
+
+    remove_original_subscription_manager()
+    install_rhel_subscription_manager()
+
+
+def remove_original_subscription_manager():
+    loggerinst = logging.getLogger(__name__)
+    loggerinst.info("Removing non-RHEL subscription-manager packages.")
+    # python3-subscription-manager-rhsm, dnf-plugin-subscription-manager, subscription-manager-rhsm-certificates, etc.
+    submgr_pkgs = pkghandler.get_installed_pkgs_w_different_fingerprint(
+        system_info.fingerprints_rhel, "*subscription-manager*")
+    if not submgr_pkgs:
+        loggerinst.info("No packages related to subscription-manager installed.")
         return
-    rpms_to_install = [os.path.join(sm_dir, x) for x in os.listdir(sm_dir)]
+    loggerinst.info("Upon continuing, we will uninstall the following subscription-manager pkgs:\n")
+    pkghandler.print_pkg_info(submgr_pkgs)
+    utils.ask_to_continue()
+    submgr_pkg_names = [pkg.name for pkg in submgr_pkgs]
+    utils.remove_pkgs(submgr_pkg_names, critical=False)
+
+
+def install_rhel_subscription_manager():
+    loggerinst = logging.getLogger(__name__)
+    loggerinst.info("Installing subscription-manager RPMs.")
+    rpms_to_install = [os.path.join(SUBMGR_RPMS_DIR, filename) for filename in os.listdir(SUBMGR_RPMS_DIR)]
     if rpms_to_install:
-        utils.install_pkgs(rpms_to_install, True)
-        loggerinst.info("RPMs installed:\n%s" % "\n".join(rpms_to_install))
-    else:
-        loggerinst.info("No RPM to be installed.")
-    return
+        _, ret_code = pkghandler.call_yum_cmd(
+            # We're using distro-sync as there might be various versions of the subscription-manager pkgs installed
+            # and we need these packages to be replaced with the provided RPMs from RHEL.
+            "install",
+            " ".join(rpms_to_install),
+            # When installing subscription-manager packages, the RHEL repos are not available yet => we need to use
+            # the repos that are available on the system
+            enable_repos=[],
+            disable_repos=[]
+        )
+        if ret_code:
+            loggerinst.critical("Failed to install subscription-manager packages."
+                                " See the above yum output for details.")
+        else:
+            loggerinst.info("Packages installed:\n%s" % "\n".join(rpms_to_install))
 
 
+def remove_subscription_manager():
+    loggerinst = logging.getLogger(__name__)
+    loggerinst.info("Removing RHEL subscription-manager packages.")
+    # python3-subscription-manager-rhsm, dnf-plugin-subscription-manager, subscription-manager-rhsm-certificates, etc.
+    submgr_pkgs = pkghandler.get_installed_pkgs_w_fingerprints("*subscription-manager*")
+    if not submgr_pkgs:
+        loggerinst.info("No packages related to subscription-manager installed.")
+        return
+    submgr_pkg_names = [pkg.pkg_obj.name for pkg in submgr_pkgs]
+    utils.remove_pkgs(submgr_pkg_names, backup=False, critical=False)
```

Comment:
@bocekm I believe that's all from your list fixed. Only concern I have is this function, should we use `call_yum_cmd` instead? `remove_pkgs` leans more to be used during conversion than rollback

Reply:
Good observation. Yes, I'd rather call the `yum remove` through `call_yum_cmd`. The additional benefit of that is that yum removes also all deps that were installed with subscription-manager and are not used by any other package.

Reply:
@bocekm ack. Ready for review